pandas/io/common.py

```diff
@@ -243,12 +228,19 @@ def get_filepath_or_buffer(
 
         return file_obj, encoding, compression, True
 
-    if isinstance(filepath_or_buffer, (str, bytes, mmap.mmap)):
-        return _expand_user(filepath_or_buffer), None, compression, False
+    if is_file_like(filepath_or_buffer) or isinstance(filepath_or_buffer, mmap.mmap):
+        return filepath_or_buffer, None, compression, False
 
-    if not is_file_like(filepath_or_buffer):
+    try:
+        return (
+            _expand_user(filepath_or_buffer, fallback=False),
+            None,
+            compression,
+            False,
+        )
+    except TypeError as e:
         msg = f"Invalid file path or buffer object type: {type(filepath_or_buffer)}"
-        raise ValueError(msg)
+        raise ValueError(msg) from e
```

Comment:
can you use "err" instead of "e"?  1-letter variable names are hard to grep for

Reply:
done !